pkg/osdCloud/aws.go

```diff
@@ -178,3 +179,93 @@ func GenerateRoleSessionName(client aws.Client) (string, error) {
 
 	return fmt.Sprintf("RH-SRE-%s", username), nil
 }
+
+func GenerateAWSClient(awsProfile string, clusterID string) (aws.Client, error) {
+
+	ocmClient := utils.CreateConnection()
+	defer ocmClient.Close()
+
+	clusterResp, err := ocmClient.ClustersMgmt().V1().Clusters().Cluster(clusterID).Get().Send()
+	if err != nil {
+		fmt.Println(err)
+		return nil, err
+	}
+	cluster := clusterResp.Body()
+	clusterRegion := cluster.Region().ID()
+
+	// Builds the base client using the provided creds (via profile or env vars)
+	awsClient, err := aws.NewAwsClient(awsProfile, clusterRegion, "")
+	if err != nil {
+		fmt.Printf("Could not build AWS Client: %s\n", err)
+		return nil, err
+	}
+
+	// Get the right partition for the final ARN
+	partition, err := aws.GetAwsPartition(awsClient)
+	if err != nil {
+		return nil, err
+	}
+
+	// Generate a session name using the SRE's kerberos ID
+	sessionName, err := GenerateRoleSessionName(awsClient)
+	if err != nil {
+		fmt.Printf("Could not generate Session Name: %s\n", err)
+		return nil, err
+	}
+
+	if cluster.CCS().Enabled() {
+		var err error
+		// Determine the right jump role
+		targetRoleArnString, err := utils.GetSupportRoleArnForCluster(ocmClient, clusterID)
+		if err != nil {
+			return nil, err
+		}
+
+		targetRoleArn, err := arn.Parse(targetRoleArnString)
+		if err != nil {
+			return nil, err
+		}
+
+		targetRoleArn.Partition = partition
+
+		// Start the jump role chain. Result should be credentials for the ManagedOpenShift Support role for the target cluster
+		assumedRoleCreds, err := GenerateSupportRoleCredentials(awsClient, targetRoleArn.AccountID, clusterRegion, sessionName, targetRoleArn.String())
+		if err != nil {
+			return nil, err
+		}
+
+		awsClient, err = aws.NewAwsClientWithInput(&aws.AwsClientInput{
+			AccessKeyID:     *assumedRoleCreds.AccessKeyId,
+			SecretAccessKey: *assumedRoleCreds.SecretAccessKey,
+			SessionToken:    *assumedRoleCreds.SessionToken,
+			Region:          clusterRegion,
+		})
+		if err != nil {
+			return nil, err
+		}
```

Comment:
I think making this into a function `BuildCCSClusterClient()` might be helpful, in case this logic ever needs to be ran on its own outside of larger function here.